appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java

```diff
@@ -46,6 +46,7 @@
   private final Provider<UserAgentPublisher> userAgentPublisherProvider;
   private final Provider<HeartBeatInfo> heartBeatInfoProvider;
   private final List<AppCheckTokenListener> appCheckTokenListenerList;
+  private final List<AppCheckListener> appCheckListenerList;
```

Comment:
It occurs to me that we could, in theory, have both of these implement an interface and then we don't have all of this tricky logic to deal with both lists everywhere. Instead, we'd have one list, and then a helper Factory class to deal with constructing the right thing to shove into each instance. 

That being said, it's not clear to me if the drawbacks to that that approach outweigh the benefits, I just wanted to flag that there was an option to keep the messiness contained to a single place if we wanted.

Reply:
I'm not sure I understand this suggestion -- do you mean have both `AppCheckListener` and `AppCheckTokenListener` implement some other `Listener` interface, and then just have a `List<Listener>` here?

The `onAppCheckTokenChanged` methods of `AppCheckListener` and `AppCheckTokenListener` take in different parameters (`AppCheckToken` vs. `AppCheckTokenResult`), so I don't think that would work.

Reply:
Right, that's what you would need a helper/factory for, do deal with parseling the token into the right format for triggering the listener. Then all of the complexity of having 2 types of listeners only lives in that one place. Whether or not that's preferable is up to you

Reply:
See comment below.